index.js

```diff
@@ -11,6 +11,23 @@ var slice = Array.prototype.slice;
 
 module.exports = co['default'] = co.co = co;
 
+// This allows a user to provide their own promise library; this is useful
+// as there are libraries which offer benefits over the ES6 Promise object
+// but are also API-compatible (or could be easily made to be so).  One
+// particular benefit of using the same promise is that some libraries
+// (notably q and bluebird) support long stack traces across multiple
+// callbacks which is useful for debugging; however, when using a different
+// promise in the middle (or particularly the end) of the chain breaks
+// this functionality.
+//
+// This is referred to in some libraries as Bring Your Own Promise (BYOP)
+//
```

Comment:
This is a minor style detail, but for the sake of style consistency, should this comment be formatted to match the style of the other multiline comments? For example:

```
/**
 *  Stuff
 */
```